app/scripts/metamask-controller.js

```diff
@@ -1825,14 +1826,20 @@ export default class MetamaskController extends EventEmitter {
    * @param {Object} msgParams - The params passed to eth_sign.
    * @param {Function} cb - The callback function called with the signature.
    */
-  newUnsignedMessage(msgParams, req) {
-    const promise = this.messageManager.addUnapprovedMessageAsync(
-      msgParams,
-      req,
-    );
-    this.sendUpdate();
-    this.opts.showUserConfirmation();
-    return promise;
+  async newUnsignedMessage(msgParams, req) {
+    const data = normalizeMsgData(msgParams.data);
```

Comment:
It looks like this `normalizeMsgData` function is being called twice now - once here, and once in `addUnapprovedMessage` in `message-manager-.js`.

I don't really follow why it was moved to be honest. Also, does it need to be called twice?

Reply:
We need to check the 32 byte length somehow, and data can be either a Buffer or a hex string. Without this function we would need to check for both types of incoming data. Normalising it is simpler

Reply:
Gotcha. I hadn't realized a length of 32 was a requirement. Testing with prod confirms that it is the case though. Mostly anyway.

I noticed one odd exception: if the message contains an odd number of digits, the last digit is ignored. I see how that would happen, as that would be 32 and a half bytes. I guess it interprets the string as bytes and ignores any remainder.

But imposing this new restriction could break sites that are unintentionally relying on their signatures working even when they add an extra byte. Perhaps we should allow the extra character for this reason, just to make this non-breaking.